dcpy/utils/s3.py

```diff
@@ -66,38 +65,38 @@ def download_file(
 ):
     """Downloads a file from S3"""
     with _progress() as progress:
-        client_ = client()
-        size = client_.head_object(Bucket=bucket, Key=key)["ContentLength"]
+        size = get_metadata(bucket, key)["content-length"]
         task = progress.add_task(
             f"[green]Downloading [bold]{Path(key).name}[/bold]", total=size
         )
-        client_.download_file(
+        client().download_file(
             Bucket=bucket,
             Key=key,
             Filename=str(path),
             Callback=lambda bytes: progress.update(task, advance=bytes),
         )
 
 
+def get_metadata(bucket: str, key: str) -> dict:
+    """Gets custom metadata as well as three standard s3 fields"""
+    response = client().head_object(Bucket=bucket, Key=key)
+    metadata = response["Metadata"]
+    metadata["last-modified"] = response["LastModified"]
+    metadata["content-length"] = response["ContentLength"]
+    metadata["content-type"] = response["ContentType"]
```

Comment:
These three are the default ones given by DO that seemed vaguely relevant. Converting to dashed because when I specify custom metadata, key is converted to all lowercase and underscores converted to dashes, so at least they'll be consistent. We can change this if it annoys us